lib/knapsack_pro/adapters/rspec_adapter.rb

```diff
@@ -127,14 +127,26 @@ def bind_save_report
       def bind_before_queue_hook
         ::RSpec.configure do |config|
           config.before(:suite) do
-            unless ENV['KNAPSACK_PRO_BEFORE_QUEUE_HOOK_CALLED']
-              ENV['KNAPSACK_PRO_BEFORE_QUEUE_HOOK_CALLED'] = 'true'
-              KnapsackPro::Hooks::Queue.call_before_queue
-            end
+            KnapsackPro::Hooks::Queue.call_before_queue
+          end
+        end
+      end
+
+      def bind_after_queue_hook
+        ::RSpec.configure do |config|
+          config.after(:suite) do
+            KnapsackPro::Hooks::Queue.call_after_queue
           end
         end
       end
 
+      def bind_queue_mode
+        # call the following method before `super` to ensure the `after_queue` hook that runs in after(:suite) is called before after(:suite) hook defined by `bind_time_tracker`
+        bind_after_queue_hook
+
+        super
```

Comment:
> make log_batch_duration use append_after instead of after

I think logging the duration of tests after each batch is unnecessary. Displaying the test duration after each batch creates just noise. I log the duration of tests only once, at the very end of the queue, to simplify the code.

change: https://github.com/KnapsackPro/knapsack_pro-ruby/pull/237/commits/1267eb1d217264d2d440f3bb2d0153e800997e20

Reply:
refactor: https://github.com/KnapsackPro/knapsack_pro-ruby/pull/237/commits/50e94afb41ac6e277aee0e0792a85a9ac384f2d2

Extracted `bind_after_queue_hook` to base adapter.